server/controllers/patients/addHistoryLog/index.js

```diff
@@ -0,0 +1,62 @@
+const { boomify, historyLogSchema } = require('../../../utils');
+
+const {
+  addHistoryLogQuery,
+  getAppointmentsByIdQuery,
+  updateAppointmentStatusQuery,
+} = require('../../../database/queries');
+
+const addHistoryLog = async (req, res, next) => {
+  try {
+    const { appointmentId } = req.params;
+    const { description, price, payment } = req.body;
+
+    const isValid = await historyLogSchema.isValid({
+      appointmentId,
+      description,
+      price,
+      payment,
+    });
+
+    if (!isValid) {
+      return next(
+        boomify(400, 'Invalid input', 'please Correct the input and try again'),
+      );
+    }
+
+    const {
+      rows: [appointmentData],
+    } = await getAppointmentsByIdQuery({ appointmentId });
+
+    if (!appointmentData) {
+      return next(
+        boomify(
+          404,
+          'appointment not found',
+          'Wrong appointment Id, check your data and try again',
+        ),
+      );
+    }
+
+    const { rows: data } = await addHistoryLogQuery({
+      appointmentId,
+      description,
+      price,
+      payment,
+    });
+
+    if (!appointmentData.is_done) {
```

Comment:
I think we don't here a condition to check is_done, it's always false and you need to change it to true

Reply:
it's not always false , so if it's true then we don't need to activate the query

Reply:
You here you want to add a log for this appointment and change it's status.
If this appointment is_done is true, that's been this appointment is already added to history,it's not correct to change it here. 

I think if is_done is true - you need to throw error